starfish/image/_stack.py

```diff
@@ -72,27 +65,46 @@ def tile_shape(self):
     def is_volume(self):
         return self._is_volume
 
-    def write(self, filepath, tile_filename_formatter=None):
-        prefix = os.path.splitext(os.path.basename(filepath))[0]
-        basepath = os.path.dirname(filepath)
-        if tile_filename_formatter is None:
-            def tile_filename_formatter(x, y, hyb, ch):
-                return "{}-x_{}-y_{}-h_{}-c_{}".format(prefix, x, y, hyb, ch)
-
-        data = list()
-
-        for hyb in range(self._num_hybs):
-            for ch in range(self._num_chs):
-                tile_filename = tile_filename_formatter(0, 0, hyb, ch)
-                tile = {
-                    'hyb': hyb,
-                    'ch': ch,
-                    'file': "{}.{}".format(tile_filename, ImageFormat.NUMPY.file_ext),
-                }
-                numpy.save(os.path.join(basepath, tile_filename), self._data[hyb, ch, :])
-                data.append(tile)
-
-        return data
+    def clone_shape(self):
+        return ImageStack(self._slicedimage.clone_shape())
+
+    def write(self, filepath, tile_opener=None):
+        seen_x_coords, seen_y_coords = set(), set()
+        for tile in self._slicedimage.get_matching_tiles():
+            seen_x_coords.add(tile.coordinates['x'])
+            seen_y_coords.add(tile.coordinates['y'])
+
+        sorted_x_coords = sorted(seen_x_coords)
+        sorted_y_coords = sorted(seen_y_coords)
+        x_coords_to_idx = {coords: idx for idx, coords in enumerate(sorted_x_coords)}
+        y_coords_to_idx = {coords: idx for idx, coords in enumerate(sorted_y_coords)}
+        # TODO: should handle Z.
```

Comment:
I would handle Z in this PR unless it's a huge nightmare. The two data types I'm working on now and next are volumetric. It would be nice to know it works! 

I have some example data that I can upload including org jsons in the _old_ specification if that gives you enough testing data.

Reply:
I think the tiles will remain 2D unless we see a compelling 3D compression format.

Therefore, making it work for Z will probably be on the rather dull side.  Will fix.

Reply:
Yep, I agree re: 2d vs 3d. Thanks, it will make it easier for the data to come. :)